### PR TITLE
refactor: add margins to headings

### PR DESCRIPTION
**Pull request by angusfretwell**

### CODE REVIEW

File: packages/components/src/Heading/__snapshots__/Heading.test.js.snap [hunk 1]
```diff
@@ -2,13 +2,18 @@
 
 exports[`<Heading /> <Heading.h1 /> renders correctly 1`] = `
 .c0 {
+  display: inline-block;
```
Comment (ehyland) [inline packages/components/src/Heading/__snapshots__/Heading.test.js.snap]: Think we want heading to be display block

Comment (angusfretwell) [reply packages/components/src/Heading/__snapshots__/Heading.test.js.snap]: Yep it is

File: packages/components/src/Heading/__snapshots__/Heading.test.js.snap [hunk 2]
```diff
@@ -32,13 +42,18 @@ exports[`<Heading /> <Heading.h2 /> renders correctly 1`] = `
...
 exports[`<Heading /> <Heading.h3 /> renders correctly 1`] = `
 .c0 {
+  display: inline-block;
+  margin: 0;
   font-size: 1.375rem;
   font-weight: 600;
   line-height: 1.5;
+  margin-bottom: 0.5rem;
+  display: block;
 }
```
[the remark this answers is not in the document]
Comment (angusfretwell) [reply packages/components/src/Heading/__snapshots__/Heading.test.js.snap]: We need to set display on it to support margin. `<Text />` is less useful without margin. We could also consider setting it to `display: block` and possibly adding an `inline` prop to make it `inline-block`

[the remark this answers is not in the document]
Comment (angusfretwell) [reply packages/components/src/Heading/__snapshots__/Heading.test.js.snap]: I think `<Heading />` extending `<Text />` makes a lot of sense.

**Pull request merged**